flowmachine/flowmachine/core/cache.py

```diff
@@ -226,6 +233,8 @@ def reset_cache(connection: "Connection", redis: StrictRedis) -> None:
     ----------
     connection : Connection
     redis : StrictRedis
+    protect_table_objects : bool, default True
+        Set to False to also remove cache metadata for Table objects which point to tables outside the cache schema
```

Comment:
Setting `protect_table_objects=False` will cause the daily_location store to fail, as before. I don't think this is a problem, since this is inevitable if we reset the cache to the state it was in before creating a query object, but it's probably worth adding a note to mention that resetting with `protect_table_objects=False` may cause existing query objects to fail to execute.

Reply:
Alternatively, I guess we could call `self._db_store_cache_metadata` from `Table.get_query`?

Reply:
Mm. That's not gonna do it unfortunately, because `get_query` will never get called for table's used in `EventTableSubset`.